data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ContainsSubstringExpressionFunction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+import java.util.List;
+import javax.inject.Named;
+import java.util.function.Function;
+
+@Named
+public class ContainsSubstringExpressionFunction implements ExpressionFunction {
+    public String getFunctionName() {
+        return "containsSubstring";
+    }
+
+    public Object evaluate(final List<Object> args, Event event, Function<Object, Object> convertLiteralType) {
+        if (args.size() != 2) {
+            throw new RuntimeException("containsSubstring() takes exactly two arguments");
+        }
+        String[] strArgs = new String[2];
+        for (int i = 0; i < 2; i++) {
+            Object arg = args.get(i);
+            if (!(arg instanceof String)) {
+                throw new RuntimeException("containsSubstring() takes only string type arguments");
+            }
+            String str = (String) arg;
+            if (str.charAt(0) == '"') {
+                strArgs[i] = str.substring(1, str.length()-1);
+            } else if (str.charAt(0) == '/') {
+                Object obj = event.get(str, Object.class);
+                if (obj == null) {
+                    return false;
+                }
+                if (!(obj instanceof String)) {
+                    throw new RuntimeException("containsSubstring() takes only string type arguments");
```

Comment:
For all of the exceptions thrown in this class, let's be more specific here and provide our users with the parameter that result in the exception. This will help them with debugging any failures in their pipeline configuration or data.

```
throw new RuntimeException(String.format("containsSubstring() takes only string type arguments. "%s" is not of type string", arg);
```